sdk/eventhubs/azure-messaging-eventhubs/README.md

```diff
@@ -1,5 +1,35 @@
 # Azure Event Hubs client library for Java
 
+- [Getting started](#getting-started)
+- [Prerequisites](#prerequisites)
+- [Adding the package to your product](#adding-the-package-to-your-product)
+- [Methods to authorize with Event Hubs](#methods-to-authorize-with-event-hubs)
+- [Create an Event Hub client using a connection string](#create-an-event-hub-client-using-a-connection-string)
+- [Create an Event Hub client using Microsoft identity platform (formerly Azure Active
+  Directory)](#create-an-event-hub-client-using-microsoft-identity-platform-formerly-azure-active-directory)
+  - [Authorizing with AAD application client secret](#authorizing-with-aad-application-client-secret)
+- [Key concepts](#key-concepts)
+- [Examples](#examples)
+- [Publish events to an Event Hub](#publish-events-to-an-event-hub)
+  - [Event Hub producer creation](#event-hub-producer-creation)
+  - [Partition identifier](#partition-identifier)
+  - [Partition key](#partition-key)
+- [Consume events from an Event Hub partition](#consume-events-from-an-event-hub-partition)
+  - [Consume events with EventHubAsyncConsumer](#consume-events-with-eventhubasyncconsumer)
+  - [Consume events with EventHubConsumer](#consume-events-with-eventhubconsumer)
+- [Consume events using an Event Processor](#consume-events-using-an-event-processor)
+- [Troubleshooting](#troubleshooting)
+- [Enable client logging](#enable-client-logging)
+- [Enable AMQP transport logging](#enable-amqp-transport-logging)
+  - [Sample "logging.config" file](#sample-loggingconfig-file)
+- [Common exceptions](#common-exceptions)
+  - [AMQP exception](#amqp-exception)
+  - [Operation cancelled exception](#operation-cancelled-exception)
+  - [Message size exceeded](#message-size-exceeded)
+- [Other exceptions](#other-exceptions)
+- [Next steps](#next-steps)
+- [Contributing](#contributing)
```

Comment:
The TOC itself is very long and now will require a reader to scroll to get to the main part of the doc.
If we do want to keep it for sure, I would suggest to keep only the top level instead of listing out all headings

- Getting started
- Key concepts
- Examples
- Troubleshooting
- Exceptions
- Next steps
- Contributing

Reply:
Oh yeah. I removed the 4th level content. I think those 7 are too broad, so having one level below it makes it easier to understand what sort of content is in side the "Getting started" section, etc.